Upgrade application from LB2 to LB3

jisoolee opened the pull request:
Update for loopback version 3

jisoolee committed:
Upgrade application from LB2 to LB3

BREAKING CHANGE: upgrade the application from LB2 to LB3 using guide
provided by https://loopback.io/doc/en/lb3/Migrating-to-3.0.html.
Upgrade dependencies accordingly.

Co-authored-by: Biniam Admikew <biniam@ca.ibm.com>

rmg reviewed (changes requested):
The dependencies are being updated to versions that require node 6+ and 8+. The `package.json` should be updated to include `"engines": {"node": ">=8"}` to reflect this so that CI knows not to even try.

bajtos reviewed (approved):
I am not familiar with this codebase. I don't see any obvious problems and the tests pass, so LGTM.

Kudos for bumping-up the major version :+1:

jisoolee committed:
chore: upgrade dependencies

upgrade dependencies and fix lint issues introduced
as part of the new version of eslint-config-strongloop
